rest-server/src/models/vc.js

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+// module dependencies
+const unirest = require('unirest');
+const yarnConfig = require('../config/yarn');
+const logger = require('../config/logger');
+
+class Vc {
+  constructor(name, next) {
+    this.getVcList((vcList, error) => {
+      if (error === null) {
+        if (name in vcList) {
+          for (let key of Object.keys(vcList[name])) {
+            this[key] = vcList[name][key];
+          }
+        } else {
+          error = new Error('VcNotFound');
+        }
+      }
+      next(this, error);
+    });
+  }
+
+  getCapacitySchedulerInfo(queueInfo) {
+    let queues = {};
+    function traverse(queueInfo, queueDict) {
+      if (queueInfo.type === 'capacitySchedulerLeafQueueInfo') {
+        queueDict[queueInfo.queueName] = {
+          capacity: queueInfo.capacity,
+          usedCapacity: queueInfo.usedCapacity,
+          maxCapacity: queueInfo.maxCapacity,
+          absoluteCapacity: queueInfo.absoluteCapacity,
+          absoluteMaxCapacity: queueInfo.absoluteMaxCapacity,
+          absoluteUsedCapacity: queueInfo.absoluteUsedCapacity,
+          maxApplications: queueInfo.maxApplications,
+          maxApplicationsPerUser: queueInfo.maxApplicationsPerUser,
+          numActiveApplications: queueInfo.numActiveApplications,
+          numApplications: queueInfo.numApplications,
+          numContainers: queueInfo.numContainers,
+          numPendingApplications: queueInfo.numPendingApplications,
+          resourcesUsed: queueInfo.resourcesUsed,
+          state: queueInfo.state,
```

Comment:
This may need more discussion. But if we use dominant calculator, how to get the number of available GPUs for specific VC? you don't know which kind of resource take the lead.

Reply:
You're correct. But I'm talking about user experience instead of implementation. Let's discuss on scrum meeting.